### DISCUSSION TITLE
Using Tanstack Router with a framework library.

### DISCUSSION THREAD

**Discussion by ScottKemp** [TanStack/router#1255] [Ideas]
I am creating multiple React applications that will use my common library (Lets call LIBX) which will create the common UI scaffholding with auth, 404, etc pages.  I also would like my application to have modules that are relatively contained.  When I was using angular, it had a way of letting child modules/libraries define their own routes, hiding the complexity of their routes from the main program.

From what I have seen from TanStack, using the code based system and not file based, is that all the routes end up in one giant file.  I am trying to avoid this.  I cant see how to let LIBX define some routes in its code, the Users module of ApplicationX maintain it's own routes, etc.

If it is possible with Tanstack, it would be nice to have and example of it.

Pseudo Code for what I am looking for:

ApplicationX - router.tsx:

    routerframework = LibX.CreateRouterFramework()
    LibX.AddLibXCommonRoutes(routerframework)
    AddUsersRoutes( routerframework )
    AddProjectRoutes( routerframework )
    AddSettingsRoutes( routerframework )
    
    LibX.CompleteRouterConstruction( routerframework );

ApplicationY - router.tsx

    routerframework = LibX.CreateRouterFramework()
    LibX.AddLibXCommonRoutes(routerframework)
    AddCarRoutes( routerframework )
    AddBoatRoutes( routerframework )
    AddSettingsRoutes( routerframework )
    
    LibX.CompleteRouterConstruction( routerframework );

**Comment by itshallrun** [comment]
Have you found a solution for this?
I have the exact same issue.

I really hope, that I won't have to migrate back to react router for that.
